File: InterviewApi/Models/PermitRegistration.cs
using System.ComponentModel.DataAnnotations;

namespace PermitApi.Models
{
    public class PermitRegistration
    {
        public int Id { get; set; }
        
        [Required]
        [StringLength(100)]
        public string ApplicantName { get; set; } = string.Empty;
        
        [Required]
        [EmailAddress]
        public string Email { get; set; } = string.Empty;
        
        [Required]
        public string PermitType { get; set; } = string.Empty;
        
        [Required]
        [StringLength(200)]
        public string ProjectDescription { get; set; } = string.Empty;
        
        [Required]
        [StringLength(300)]
        public string Address { get; set; } = string.Empty;
        
        [Range(1, 1000000)]
        public decimal EstimatedCost { get; set; }
        
        public DateTime SubmissionDate { get; set; } = DateTime.UtcNow;
        
        public PermitStatus Status { get; set; } = PermitStatus.Pending;
        
        public string? ReviewNotes { get; set; }
    }
    
    public enum PermitStatus
    {
        Pending,
        UnderReview,
        Approved,
        Rejected,
        RequiresMoreInfo
    }
    
    public enum PermitType
    {
        Building,
        Electrical,
        Plumbing,
        Mechanical,
        Demolition,
        SignPermit
    }
}

File: interview-frontend/src/app/models/permit-registration.ts
export interface PermitRegistration {
  id: number;
  applicantName: string;
  email: string;
  permitType: string;
  projectDescription: string;
  address: string;
  estimatedCost: number;
  submissionDate: string;
  status: PermitStatus;
  reviewNotes?: string;
}

export interface CreatePermitRegistration {
  applicantName: string;
  email: string;
  permitType: string;
  projectDescription: string;
  address: string;
  estimatedCost: number;
}

export enum PermitStatus {
  Pending = 0,
  UnderReview = 1,
  Approved = 2,
  Rejected = 3,
  RequiresMoreInfo = 4,
}

export const PermitTypes = [
  'Building',
  'Electrical',
  'Plumbing',
  'Mechanical',
  'Demolition',
  'SignPermit',
];
